packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -124,6 +124,8 @@ export interface LightningElementConstructor {
     new (): LightningElement;
     readonly prototype: LightningElement;
     readonly CustomElementConstructor: HTMLElementConstructor;
+
+    delegatesFocus?: boolean;
 }
 
 export declare let LightningElement: LightningElementConstructor;
```

Comment:
Lifting `LightningElementConstructor` and `LightningElement` type definition to a separate module will reduce the cyclic dependencies between modules quite a bit.